5-AdvancedScenarios/1-call-graph-bff/README.md

```diff
@@ -0,0 +1,447 @@
+
+> :information_source: To learn how applications integrate with [Microsoft Graph](https://aka.ms/graph), consider going through the recorded session: [An introduction to Microsoft Graph for developers](https://www.youtube.com/watch?v=EBbnpFdB92A)
+
+## Scenario
+
+1. The client-side React SPA initiates token acquisition by calling the login endpoint of the Express web app.
+1. Express web app uses **MSAL Node** to sign-in a user and obtain a JWT [ID Token](https://aka.ms/id-tokens) and an [Access Token](https://aka.ms/access-tokens) from **Azure AD**.
+1. Express web app uses the **access token** as a *bearer* token to authorize the user to call the Microsoft Graph API protected by **Azure AD**.
+1. Express web app returns the Microsoft Graph `/me` endpoint response back to the React SPA.
+
+![Scenario Image](./ReadmeFiles/sequence.png)
+
+## Contents
+
+| File/folder                      |                                Description                                 |
+|--------------------------------- |----------------------------------------------------------------------------|
+| `app.js`                         | Application entry point.                                                   |
+| `authConfig.js`                  | Contains authentication configuration parameters.                          |
+| `auth/AuthProvider.js`           | A custom wrapper around MSAL Node for authN/authZ.                         |
+| `utils/graphClient`              | Instantiates Graph SDK client using a custom authentication provider.      |
+| `client/src/context/AuthContext` | React context to fetch user information from the backend.                  |
+| `client/src/pages/Profile`       | Calls `auth/profile` route on backend to call the Microsoft Graph `/me` endpoint   |
+
+## Prerequisites
+
+* [Node.js](https://nodejs.org/en/download/) must be installed to run this sample.
+* [Visual Studio Code](https://code.visualstudio.com/download) is recommended for running and editing this sample.
+* [VS Code Azure Tools](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-node-azure-pack) extension is recommended for interacting with Azure through VS Code Interface.
+* A modern web browser.
+* An **Azure AD** tenant. For more information, see: [How to get an Azure AD tenant](https://docs.microsoft.com/azure/active-directory/develop/test-setup-environment#get-a-test-tenant)
+* A user account in your **Azure AD** tenant.
+
+## Setup the sample
+
+### Step 1: Clone or download this repository
+
+From your shell or command line:
+
+```console
+git clone https://github.com/Azure-Samples/ms-identity-javascript-nodejs-tutorial.git
+```
+
+or download and extract the repository *.zip* file.
+
+> :warning: To avoid path length limitations on your operating systems, we recommend cloning into a directory near the root of your drive.
```

Comment:
```suggestion
> :warning: To avoid path length limitations on your operating system, we recommend cloning into a directory near the root of your drive.
```